--- plymouth-0.8.4/src/plugins/renderers/drm/ply-renderer-nouveau-driver.c~	2010-08-19 19:01:38.000000000 +0200
+++ plymouth-0.8.4/src/plugins/renderers/drm/ply-renderer-nouveau-driver.c	2012-05-18 13:07:11.302627528 +0200
@@ -41,9 +41,8 @@
 #include <unistd.h>
 
 #include <drm.h>
+#include <nouveau.h>
 #include <nouveau_drm.h>
-#include <nouveau_drmif.h>
-#include <nouveau_bo.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
 
