--- plymouth-0.8.6.1/systemd-units/plymouth-start.service.in.orig	2012-06-20 17:57:03.000000000 +0200
+++ plymouth-0.8.6.1/systemd-units/plymouth-start.service.in	2012-07-27 18:52:25.292260993 +0200
@@ -9,5 +9,5 @@
 
 [Service]
 ExecStart=@PLYMOUTH_DAEMON_DIR@/plymouthd --mode=boot --pid-file=@plymouthruntimedir@/pid --attach-to-session
-ExecStartPost=-/bin/udevadm settle --timeout=30 --exit-if-exists=/sys/class/drm/card0/dev;-/bin/udevadm settle --timeout=30 --exit-if-exists=/sys/class/graphics/fb0/dev; -@PLYMOUTH_CLIENT_DIR@/plymouth show-splash
+ExecStartPost=-/sbin/udevadm settle --timeout=30 --exit-if-exists=/sys/class/drm/card0/dev;-/sbin/udevadm settle --timeout=30 --exit-if-exists=/sys/class/graphics/fb0/dev; -@PLYMOUTH_CLIENT_DIR@/plymouth show-splash
 Type=forking
