--- plymouth-0.9.5/src/libply-splash-graphics/Makefile.am.orig	2020-06-15 16:16:47.000000000 +0200
+++ plymouth-0.9.5/src/libply-splash-graphics/Makefile.am	2020-07-20 22:30:24.853900043 +0200
@@ -26,7 +26,7 @@
                                    -DPLYMOUTH_BACKGROUND_END_COLOR=$(background_end_color) \
                                    -DPLYMOUTH_BACKGROUND_START_COLOR=$(background_start_color) \
                                    -DPLYMOUTH_PLUGIN_PATH=\"$(PLYMOUTH_PLUGIN_PATH)\"
-libply_splash_graphics_la_LIBADD = $(PLYMOUTH_LIBS) $(IMAGE_LIBS) ../libply/libply.la ../libply-splash-core/libply-splash-core.la
+libply_splash_graphics_la_LIBADD = ../libply/libply.la ../libply-splash-core/libply-splash-core.la $(PLYMOUTH_LIBS) $(IMAGE_LIBS)
 libply_splash_graphics_la_LDFLAGS = -export-symbols-regex '^[^_].*' \
                                     -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
                                     -no-undefined
