--- plymouth-0.9.2/src/plugins/splash/tribar/plugin.c.orig	2015-05-24 21:32:03.443625675 +0200
+++ plymouth-0.9.2/src/plugins/splash/tribar/plugin.c	2015-05-24 21:35:09.136951216 +0200
@@ -180,10 +180,10 @@
                                           0xffffff);
         ply_terminal_set_color_hex_value (terminal,
                                           PLY_TERMINAL_COLOR_BLUE,
-                                          0x0073B3);
+                                          PLYMOUTH_BACKGROUND_START_COLOR);
         ply_terminal_set_color_hex_value (terminal,
                                           PLY_TERMINAL_COLOR_BROWN,
-                                          0x00457E);
+                                          PLYMOUTH_BACKGROUND_END_COLOR);
 
         ply_text_display_set_background_color (view->display,
                                                PLY_TERMINAL_COLOR_BLACK);
--- plymouth-0.9.2/src/plugins/splash/text/plugin.c.orig	2015-05-24 21:32:03.446959010 +0200
+++ plymouth-0.9.2/src/plugins/splash/text/plugin.c	2015-05-24 21:36:36.543614214 +0200
@@ -171,13 +171,13 @@
 
         ply_terminal_set_color_hex_value (terminal,
                                           PLY_TERMINAL_COLOR_BLACK,
-                                          0x2e3436);
+                                          PLYMOUTH_BACKGROUND_START_COLOR);
         ply_terminal_set_color_hex_value (terminal,
                                           PLY_TERMINAL_COLOR_WHITE,
                                           0xffffff);
         ply_terminal_set_color_hex_value (terminal,
                                           PLY_TERMINAL_COLOR_BROWN,
-                                          0x979a9b);
+                                          PLYMOUTH_BACKGROUND_END_COLOR);
 
         ply_text_display_set_background_color (view->display,
                                                PLY_TERMINAL_COLOR_BLACK);
